fluidreview/api.py

```diff
@@ -235,3 +236,29 @@ def list_users():
             yield fluid_user
         next_page = urlparse(response['next']).query
         url = 'users?{}'.format(next_page) if next_page else None
+
+
+def post_payment(order):
+    """
+    Update amount paid by a user for a class when an order is fulfilled.
+    """
+    if order.status != Order.FULFILLED:
+        return
+    klass = order.get_klass()
+    user = order.user
+    if not klass or klass.bootcamp.legacy:
+        return
+    total_paid = Line.total_paid_for_klass(order.user, klass.klass_key).get('total') or Decimal('0.00')
+    payment_metadata = {
+        'value': '{:0.2f}'.format(total_paid)
+    }
+    webhook = WebhookRequest.objects.filter(user_id=user.profile.fluidreview_id, award_id=klass.klass_key).last()
+    response = FluidReviewAPI().put(
+        'submissions/{}/metadata/{}/'.format(webhook.submission_id, settings.FLUIDREVIEW_AMOUNTPAID_ID),
```

Comment:
Can you add a check if `webhook` is `None` before this line?

Reply:
I was thinking it would be okay for an exception to be thrown on the next line if it is None, because this shouldn't happen?  Or is it okay for a non-legacy klass to not have any WebhookRequests when a payment is submitted?

Reply:
I think I meant to ask for a check for `webhook.submission_id`. If `webhook` was `None` then we would get an exception here which has an obvious source. However if `webhook.submission_id` was none it would be `submissions/None/metadata/{}/` which would be confusing to diagnose

Reply:
✅ with test